Firmware/Configuration.c

```diff
@@ -0,0 +1,12 @@
+#include <stdint.h>
+#include <avr/pgmspace.h>
+//!?! #include "Configuration.h"
+#include "Configuration_prusa.h"
+
+const uint16_t _nPrinterType PROGMEM=PRINTER_TYPE;
+const char _sPrinterName[] PROGMEM=PRINTER_NAME;
+const uint16_t _nPrinterMmuType PROGMEM=PRINTER_MMU_TYPE;
+const char _sPrinterMmuName[] PROGMEM=PRINTER_MMU_NAME;
+
```

Comment:
don't use a C file when not necessary. Rename to Configuration.cpp and you won't have problems linking these variables without "extern "C"".